pkg/getter/getter.go

```diff
@@ -24,6 +24,19 @@ import (
 	"helm.sh/helm/pkg/cli"
 )
 
+// Options are generic parameters to be provided to the getter during instantiation.
+//
+// Getters may or may not ignore these parameters as they are passed in.
+type Options struct {
```

Comment:
Is there a reason why we aren't using functional options instead for this? I always worry about using "option" or "config" structs because they can balloon fairly quickly and become unwieldy

Reply:
Do you have an example? Open to opinions here (or... options... as it were :wink:)

Reply:
Ah nvm I found an article. Haven't heard that term before. I believe we did this before with the Helm gRPC client as well. Will take a look at that approach.

Reply:
okay, implemented. Let me know what you think.

Reply:
I am still a little bit unsure of having a backing `options` struct, but I can't think of an easy way to make it work for both types of getters without having duplicate options for each one. It is abstracted away from the user though because only the options funcs are exported, so this should work just fine. We can always change the backing later because it is unexported and we won't break backwards compatibility